src/BaseCollection.php

```diff
@@ -89,7 +89,7 @@ public function column(string $column, $index_key = null): array
 	 */
 	public function map(callable $callback): BaseCollection
 	{
-		return new static(array_map($callback, $this->getArrayCopy()), $this->getTotalCount());
+		return new self(array_map($callback, $this->getArrayCopy()), $this->getTotalCount());
```

Comment:
This is wrong, as it will return a `BaseCollection` instance when an inherited class instance is expected instead. If the return type hint allows for `static`, then it should be used instead.

Reply:
You are right. I wasn't aware that this class would also be extended. The problem with using `static` is that the extending class could can overwrite the `__construct()` method with different parameters. See https://phpstan.org/blog/solving-phpstan-error-unsafe-usage-of-new-static

I will mute this error by using `get_class($this);` for now but this has to be refactored in the future.

Reply:
I saw this, and in one instance we are extending the constructor, so we can't use any of the remediation solutions. It's another one of these "it might break if" issues that currently don't apply to us that we unfortunately have to mute rather than "fix".

Reply:
I'm also curious about the different between `static` and `get_class($this)` in this case, why does it mute this warning?

Reply:
> I'm also curious about the different between `static` and `get_class($this)` in this case, why does it mute this warning?

I cannot say for sure but at level 3 PHPStan will start again complaining about this code.